api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -196,6 +196,14 @@ message ExternalProcessor {
   // will be added to StreamInfo's filter state under the namespace corresponding to the
   // ext_proc filter name.
   google.protobuf.Struct filter_metadata = 13;
+
+  // [#not-implemented-hide:]
+  // If ``allow_mode_override`` is set to true, the filter config :ref:`processing_mode
+  // <envoy_v3_api_field_extensions.filters.http.ext_proc.v3.ExternalProcessor.processing_mode>`
+  // can be overridden by the response message from the external processing server
+  // :ref:`mode_override <envoy_v3_api_field_service.ext_proc.v3.ProcessingResponse.mode_override>`.
+  // If not set, ``mode_override`` API in the response message will be ignored.
+  bool allow_mode_override = 14;
```

Comment:
as mentioned offline, given that this API is alpha and not declared stable, I think we should make the default behavior be the same one, i.e. envoy rejects mode overrides that don't match with the processing mode it has. The only mode override that ext_proc server can do is turn off processing modes but not enable new processing modes. This feels like the more intuitive and simpler model that people may use for optimization purposes.

Those who want to turn on new modes from ext_proc server will need to introduce the api option and do the actual work to ensure Envoy doesn't crash as captured by the fuzzer. This can be done at later date.

Reply:
Thanks for the comments!
If ext_proc server want to tell ext_proc filter to not send any further data up, there are multiple ways to do it. One way is just close the gRPC stream as here, which will set processing_complete_ to true:
https://github.com/envoyproxy/envoy/blob/fb61801c03c544968606e97303439ebcade240bd/source/extensions/filters/http/ext_proc/ext_proc.cc#L701

Overall,  since there are quite a few combinations for ProcessingMode override, like request/response header/data/trailer, NONE/SEND/SKIP NONE/BUFFERED/BUFFERED_PARTIAL/STREAMING, it will be very hard for Envoy to define what's the right mode override settings, hence accept; and what's invalid mode override, hence ignore.  What Envoy can do is to set this control API in filter config, for example, set it to true for trusted external service, and set it to false for untrusted. Thus take the accept or ignore actions.